lib/promoted/ruby/client.rb

```diff
@@ -41,22 +44,23 @@ def initialize (params={})
 
           @delivery_timeout_millis = params[:delivery_timeout_millis] || DEFAULT_DELIVERY_TIMEOUT_MILLIS
           @metrics_timeout_millis  = params[:metrics_timeout_millis] || DEFAULT_METRICS_TIMEOUT_MILLIS
-          @api_key                 = params[:api_key] || nil
 
-          @http_client = AsyncHTTPClient.new
+          @http_client = FaradayHTTPClient.new
+          @pool = Concurrent::CachedThreadPool.new
```

Comment:
How much overhead is this?

Do we want any customization to this?

Reply:
If it's a singleton (and probably even if it's not), it's quite minimal. Cached thread pool grows and shrinks the pools based on usage. I expect that yes, we want to customize it (which means changing to a different thread pool class in ruby-concurrent), but I want to re-evaluate how this class is created because it already takes a lot of params.